libraries/Microsoft.Bot.Builder.Prompts/BasePrompt.cs

```diff
@@ -7,7 +7,49 @@
 
 namespace Microsoft.Bot.Builder.Prompts
 {
+    public enum RecognitionStatus
+    {
+        /// <summary>
+        /// The data type was not recognized at all
+        /// </summary>
+        NotRecognized,
+
+        /// <summary>
+        /// Data type was recognized and validated
+        /// </summary>
+        Recognized,
+
+        /// <summary>
+        /// The validation failed because too small
+        /// </summary>
+        TooSmall,
+
+        /// <summary>
+        /// The validation failed because too big
+        /// </summary>
+        TooBig,
+
+        /// <summary>
+        /// The validation failed because it was out of range
+        /// </summary>
+        OutOfRange
+    }
+
+    public class RecognitionResult
```

Comment:
Prompts should have `PromptResult`

Recognizers should have `RecognizerResult`